sdk/identity/azure-identity/test/perf/inc/azure/identity/test/client_certificate_credential_test.hpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @file
+ * @brief Test the overhead of authenticating with client certificate credential.
+ *
+ */
+
+#pragma once
+
+#include <azure/identity.hpp>
+#include <azure/perf.hpp>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace Azure { namespace Identity { namespace Test {
+
+  /**
+   * @brief A test to measure the authentication token performance.
+   *
+   */
+  class ClientCertificateCredentialTest : public Azure::Perf::PerfTest {
+  private:
+    std::string m_tenantId;
+    std::string m_clientId;
+    std::string m_certPath;
+    Core::Credentials::TokenRequestContext m_tokenRequestContext;
+    std::unique_ptr<Azure::Core::Credentials::TokenCredential> m_credential;
+
+  public:
+    /**
+     * @brief Setup the test.
+     *
+     */
+    void Setup() override
+    {
+      m_tenantId = m_options.GetMandatoryOption<std::string>("TenantId");
+      m_clientId = m_options.GetMandatoryOption<std::string>("ClientId");
+      m_certPath = m_options.GetMandatoryOption<std::string>("CertPath");
+      m_tokenRequestContext.Scopes.push_back(m_options.GetMandatoryOption<std::string>("Scope"));
+      if (!m_options.GetOptionOrDefault<bool>("Cache", false))
+      {
+        m_tokenRequestContext.MinimumExpiration = std::chrono::hours(1000000);
+      }
+      m_credential = std::make_unique<Azure::Identity::ClientCertificateCredential>(
+          m_tenantId,
+          m_clientId,
+          m_certPath,
+          InitClientOptions<Azure::Core::Credentials::TokenCredentialOptions>());
+    }
+
+    /**
+     * @brief Construct a new ClientCertificateCredentialTest test.
+     *
+     * @param options The test options.
+     */
+    ClientCertificateCredentialTest(Azure::Perf::TestOptions options) : PerfTest(options) {}
+
+    /**
+     * @brief Define the test
+     *
+     * @param context The cancellation token.
+     */
+    void Run(Azure::Core::Context const& context) override
+    {
+      auto t = m_credential->GetToken(m_tokenRequestContext, context);
+    }
```

Comment:
How do we run this test locally (what commands do we use) and can you please share some perf test run results?

How's the overhead here that we are measuring?

Reply:
For example, do we pass in the path to the checked-in `perf.combo.pem` file as a `CertPath` option when running this test?

Reply:
the command is documented in the Get test option method

Reply:
For information on how to run performance tests, see [the documentation](https://github.com/Azure/azure-sdk-for-cpp/blob/azure-core_1.11.2/doc/PerformanceTesting.md).